### PR TITLE
Pipeline-ify upload API

### PR DESCRIPTION
**Pull request by halfdan**
refs #5508

### CODE REVIEW

File: core/server/api/upload.js [hunk 1]
```diff
@@ -5,6 +6,15 @@ var config  = require('../config'),
...
     i18n    = require('../i18n'),
+=======
+var config   = require('../config'),
+    Promise  = require('bluebird'),
+    fs       = require('fs-extra'),
+    storage  = require('../storage'),
+    errors   = require('../errors'),
+    utils    = require('./utils'),
+    pipeline = require('../utils/pipeline'),
+>>>>>>> ace5c06... Pipeline-ify configuration/themes/upload APIs
```
Comment (novaugust) [inline core/server/api/upload.js]: lol nub

File: core/server/api/upload.js [hunk 2]
```diff
@@ -23,27 +24,42 @@ upload = {
...
         }
 
-        // Check if the file is valid
-        if (!utils.checkFileIsValid(options.uploadimage, config.uploads.contentTypes, config.uploads.extensions)) {
-            return Promise.reject(new errors.UnsupportedMediaTypeError(i18n.t('errors.api.upload.pleaseSelectValidImage')));
+        function storeUpload(options) {
+            var store = storage.getStorage(),
+                filepath = options.data.uploadimage.path;
+
+            return store.save(options.data.uploadimage).then(function (url) {
```
Comment (jaswilli) [inline core/server/api/upload.js]: `options.data`? Pretty sure this is busted and the integration tests for this api method are junk since they stub out the entire storage module.

tl;dr :sob:

**Comment by ErisDS** [comment]
I may have lead you up the garden path with this one @halfdan sorry :( I though it hadn't changed.

**Comment by ErisDS** [comment]
#6650 is merged which fixes the tests for the uploader API. That should make getting this working a bit easier.

**Comment by ErisDS** [comment]
This was a fail on my part... will try again some other time.
